workflows/secondpass_quantify_lr_rna/combine_gtf_pipeline.wdl

```diff
@@ -0,0 +1,541 @@
+            line = substr(line, 1, RSTART-1) replacement substr(line, RSTART+RLENGTH)
+            print "[DEBUG] Replacing transcript_id:", tid, "->", map[tid] > "/dev/stderr"
+        } else {
+            print "[DEBUG] transcript_id not in map:", tid > "/dev/stderr"
+            break
+        }
+    }
+    print line
+}
+' gffcomp_out.tracking gffcomp_out.combined.gtf > renamed.gtf
+
+
+  >>>
+
+  output {
+    File combined_gtf = "renamed.gtf"  # Combined GTF from all samples
+    File tracking_file = "gffcomp_out.newtracking"     # Tracking file showing transcript relationships
+    File stats_file = "gffcomp_out.stats"           # Statistics about the comparison
+  }
+
+  runtime {
+    cpu: 2
+    memory: "64G"
+    disks: "local-disk 500 SSD"
+    docker: docker_image
+  }
+}
+
+# Step 2 Task - SQANTI3 Quality Control and Classification
+task run_sqanti3 {
+  input {
+    String sample_id              # Sample identifier
+    File isoquant_gtf             # Combined GTF from gffcompare
+    File ref_annotation_gtf       # Reference annotation GTF
+    File ref_fasta                # Reference genome FASTA
+    String docker_image           # SQANTI3 docker image name
+    String docker_image_hash_or_tag # SQANTI3 docker image tag or hash
+    Int cpu = 2                   # CPU cores to use
+    Int mem_gb = 8                # Memory in GB
+    Int preemptible = 2           # Number of preemptible attempts
+    Int max_retries = 1           # Maximum number of retries
+    Int additional_disk_gb = 0    # Additional disk space in GB
+  }
+
+  # Calculate required disk space based on input file sizes
+  Int disk_space = (
+    ceil(
+      size(isoquant_gtf, "GiB") * 3 + size(ref_annotation_gtf, "GiB") + size(ref_fasta, "GiB")
+    ) + 20 + additional_disk_gb
+  )
+
+  command <<<
+    set -euo pipefail
+
+    awk '{ if ($7 != ".") print }' ~{isoquant_gtf} > "~{sample_id}.unzipped.gtf"
+
+    python /usr/local/src/SQANTI3-5.3.6/sqanti3_qc.py \
+      --report both \
+      --output ~{sample_id} \
+      ~{sample_id}.unzipped.gtf \
+      ~{ref_annotation_gtf} \
+      ~{ref_fasta} 
+  >>>
+
+  output {
+    File sq_class = "~{sample_id}_classification.txt"           #Transcript classification
+  }
+
+  runtime {
+    docker: "~{docker_image}~{docker_image_hash_or_tag}"
+    memory: mem_gb + " GiB"
+    disks: "local-disk " + disk_space + " SSD"
+    preemptible: preemptible
+    maxRetries: max_retries
+    cpu: cpu
+  }
+
+  meta {
+    allowNestedInputs: true
+  }
+}
+
+# Step 3 Tasks
+task gffread {
+  input {
+    String sample_id
+    File referenceGenome
+    File gencode_gtf
+    File referenceAnnotation_full
+    File squanti_classification
+    File updated_tracking
+    String prefix 
+    Int cpu = 2
+    Int memoryGB = 8
+    Int diskSizeGB = 20
+    String docker_image = "docker.io/hannharris/salmon-with-pandas@sha256:29adeea24ffb7d396e06c0bd7fd3c7ad258d65bd8fb9188f59e2c536023446f3"
```

Comment:
I'll probably extract the python code into a self-contained image. No need to change anything here.